### PR TITLE
HW#6 - Improved routing

### PR DESCRIPTION
**Pull request by hannapereverzieva**
Added a progress spinner on posts loading from back end, added not found page component.
Screenshot 2021-12-30 at 13 16 24

Screenshot 2021-12-30 at 13 19 20

Screenshot 2021-12-30 at 13 20 20
Screenshot 2021-12-30 at 13 21 09

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -4,13 +4,15 @@ import { PostsComponent } from "./posts/posts.component";
...
 
 const routes: Routes = [
     {path: '', redirectTo: '/', pathMatch: 'full'},
     {path: '', component: PostsComponent},
     {path: 'post/:id', component: PostPageComponent },
     {path: 'login', component: LoginComponent},
-    {path: 'create', component: CreatePageComponent}
+    {path: 'create', component: CreatePageComponent},
+    { path: '**', component: NotFoundComponent}
 ];
```
Comment (MMaksymivUa) [inline src/app/app-routing.module.ts:16]: use linter for project/or add a formatter on file save

File: src/app/app.module.ts [hunk 2]
```diff
@@ -43,11 +48,20 @@ import { HttpClientModule } from "@angular/common/http";
...
     HttpClientModule
   ],
   exports: [
     AppComponent,
   ],
+  providers: [
+      LoadingService,
+      {
+        provide: HTTP_INTERCEPTORS,
+        useClass: NetworkInterceptor,
```
Comment (MMaksymivUa) [inline src/app/app.module.ts:61]: bad name for spinner interceptor

File: src/app/app.module.ts [hunk 3]
```diff
@@ -43,11 +48,20 @@ import { HttpClientModule } from "@angular/common/http";
     ReactiveFormsModule,
     MatFormFieldModule,
     MatInputModule,
+    MatProgressSpinnerModule,
```
Comment (MMaksymivUa) [inline src/app/app.module.ts:51]: still - add a separate shared module with material components

**Comment by MMaksymivUa** [comment]
Good job!

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (hannapereverzieva) [reply src/app/app-routing.module.ts:16]: Added prettier config in this PR: 
[PR10- Enhance app](https://github.com/hannapereverzieva/blog/pull/10)

File: src/app/app.module.ts (hunk 2 above)
Comment (hannapereverzieva) [reply src/app/app.module.ts:61]: Ok, suggest the name SpinnerInterceptor.

**Pull request merged**